src/base/NekRSSeparateDomainProblem.C

```diff
@@ -0,0 +1,343 @@
+/********************************************************************/
+/*                  SOFTWARE COPYRIGHT NOTIFICATION                 */
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2021 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#include "NekRSSeparateDomainProblem.h"
+#include "Moose.h"
+#include "AuxiliarySystem.h"
+#include "TimeStepper.h"
+#include "NekInterface.h"
+#include "TimedPrint.h"
+#include "MooseUtils.h"
+#include "CardinalUtils.h"
+
+#include "nekrs.hpp"
+#include "nekInterface/nekInterfaceAdapter.hpp"
+
+registerMooseObject("CardinalApp", NekRSSeparateDomainProblem);
+
+bool NekRSSeparateDomainProblem::_first = true;
+
+InputParameters
+NekRSSeparateDomainProblem::validParams()
+{
+  InputParameters params = NekRSProblemBase::validParams();
+  params.addParam<bool>("minimize_transfers_in", false, "Whether to only synchronize nekRS "
+    "for the direction TO_EXTERNAL_APP on multiapp synchronization steps");
+  params.addParam<bool>("minimize_transfers_out", false, "Whether to only synchronize nekRS "
+    "for the direction FROM_EXTERNAL_APP on multiapp synchronization steps");
+  params.addParam<bool>("moving_mesh", false, "Moving mesh");
+  params.addParam<bool>("toNekRS_interface", false, "External app -> NekRS interface present?");
+  params.addParam<bool>("toNekRS_temperature", false, "External app -> NekRS temperature transfer?");
+  params.addParam<bool>("fromNekRS_interface", false, "NekRS -> external app interface present?");
+  params.addParam<bool>("fromNekRS_temperature", false, "NekRS -> external app temperature transfer?");
+  params.addRequiredParam<std::vector<FileName>>("ExternalApp_filename", "External app input file name");
+  MooseEnum app_type("SamApp THMApp");
+  params.addRequiredParam<MooseEnum>("ExternalApp_type", app_type, "External app type");
+  params.addParam<std::vector<int>>("fromNekRS_boundary", "Boundary ID through which nekRS will be coupled to external app");
```

Comment:
What if the user enters a boundary ID that doesn't exist? Check out the `nekrs::mesh::validBoundaryIDs` function, and we can add an error check that the boundaries here are valid.